refactor(encryptor): align Encryptor methods to match @metamask/browser-passworder

ccharly opened the pull request:
## **Description**

Align `Encryptor` methods to match `@metamask/browser-passworder` as much as possible.

## **Related issues**

Fixes https://github.com/MetaMask/accounts-planning/issues/379
 
## **Manual testing steps**

### Update app version and verify secrets

1. Create a new wallet using the last version of the app
2. Lock and unlock the app
3. Update the wallet to the code in this branch
4. Lock and unlock the app
5. Add 2-3 accounts
6. Lock and unlock the app
8. Verify the private for a random account is still accessible
9. Verify the SRP for the wallet is still accessible
10. Connect a hardware wallet and import accounts
11. Lock and unlock the app
12. Verify the state of the app is consistent

## **Screenshots/Recordings**

Not applicable

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

ccharly committed:
refactor(encryptor): align Encryptor to match browser-passworder interface

ccharly committed:
refactor(encryptor): adapt types + use types from browser-passworder

ccharly committed:
refactor(encryptor): *DERIVATION_PARAMS* -> *DERIVATION_OPTIONS*

ccharly committed:
feat(encryptor): abstract underlying encryption library + fix key parameter

There was also an error in the way we were using the actual key value.
We were passing an EncryptionKey to the native library... This would
have probably crash with the real library.

ccharly committed:
refactor(encryptor): export generateSalt

ccharly committed:
fix(encryptor): prevent mocks from being discarded

ccharly committed:
feat(encryptor): checks for more preconditions in encryption lib layer

ccharly committed:
feat(encryptor): add exportable attribute for EncryptionKey + keyFromPassword

ccharly committed:
feat(encryptor): add {import,export}Key methods

ccharly committed:
refactor(encryptor): use same test style for it.each

ccharly committed:
chore: lint

danroc commented on app/core/Encryptor/Encryptor.ts:
```diff
@@ -1,19 +1,17 @@
-  SHA256_DIGEST_LENGTH,
   ENCRYPTION_LIBRARY,
+  LEGACY_DERIVATION_OPTIONS,
   KeyDerivationIteration,
```
The linter is failing on this line:

> 'KeyDerivationIteration' is defined but never used

ccharly replied:
Yup indeed, should be fixed now! :)

ccharly committed:
chore: lint

ccharly committed:
docs: remove @throws on Encryptor's constructor

Co-authored-by: Gustavo Antunes <17601467+gantunesr@users.noreply.github.com>

ccharly committed:
Merge branch 'main' into refactor/implements-generic-encryptor

gantunesr committed:
Merge branch 'main' into refactor/implements-generic-encryptor

ccharly committed:
Merge branch 'main' into refactor/implements-generic-encryptor

ccharly committed:
refactor(encryptor): remove @ts-expect-error for crypto.getRandomValues

This is to follow those changes introduced on the mainline:
- https://github.com/MetaMask/metamask-mobile/commit/6cc76b292a39def53a97a217627851cfaf4c26aa

ccharly committed:
chore(encryptor): wording
